### PR TITLE
[GLF-4] API calls v2

### PR DESCRIPTION
**Pull request by tienviet10**

### CODE REVIEW

File: src/app/services/usergames/useEditUserGame.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import { CustomAxiosResponse, ErrorResponse } from '@app/constants/global/types';
+import { userGameRoute } from '@app/constants/global/urls';
+import client from '@app/utils/authApi';
+import { useMutation } from '@tanstack/react-query';
+import { useCallback, useMemo } from 'react';
```
Comment (changLiCoding) [inline src/app/services/usergames/useEditUserGame.ts]: Nitpick: we may want to import dependencies first and then hook and component.

File: src/app/services/usergames/useGetUserGame.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import { useQuery } from '@tanstack/react-query';
+import { CustomAxiosResponse, ErrorResponse } from '@app/constants/global/types';
+import client from '@app/utils/authApi';
+import { userGameRoute } from '@app/constants/global/urls';
+import { useCallback, useMemo } from 'react';
```
Comment (changLiCoding) [inline src/app/services/usergames/useGetUserGame.ts]: Same here

File: src/app/services/usergames/useGetUserGames.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { useQuery } from '@tanstack/react-query';
+import { CustomAxiosResponse, ErrorResponse, UserGamesByStatus } from '@app/constants/global/types';
+import client from '@app/utils/authApi';
+import { getUserGamesByStatus } from '@app/constants/global/urls';
+import { useCallback, useMemo } from 'react';
```
Comment (changLiCoding) [inline src/app/services/usergames/useGetUserGames.ts]: same as well.

File: src/app/services/usergames/useRemoveUserGame.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import { userGameRoute } from '@app/constants/global/urls';
+import client from '@app/utils/authApi';
+import { CustomAxiosResponse, ErrorResponse } from '@app/constants/global/types';
+import { useMutation } from '@tanstack/react-query';
```
Comment (changLiCoding) [inline src/app/services/usergames/useRemoveUserGame.ts:3]: same here

**Review by changLiCoding — APPROVED** [review]

File: src/app/services/usergames/useEditUserGame.ts (hunk 1 above)
Comment (tienviet10) [reply src/app/services/usergames/useEditUserGame.ts]: Fixed. I will look into the automatic sort so we dont have to worry about this later on when coding.

**Pull request merged**